### PR TITLE
Marge hw2 into main

### PR DESCRIPTION
**Pull request by VladyslavPolushyn**

### CODE REVIEW

File: hw2/assets/js/fadeIn.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// function onEntry(entry) {
```
Comment (Maxim-Bondar) [inline hw2/assets/js/fadeIn.js:1]: remove useless file

File: hw2/assets/js/mainMenu.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+mobileMenuLinks.forEach(elem => {
+	elem.addEventListener('click', () => {
+		removeActiveClass(mobileMenuLinks);
+		elem.classList.add('active');
+	})
+});
+
+function removeActiveClass(arr) {
+
```
Comment (Maxim-Bondar) [inline hw2/assets/js/mainMenu.js:19]: remove empty line

File: hw2/assets/js/slickSlider.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
+$(document).ready(function(){
```
Comment (Maxim-Bondar) [inline hw2/assets/js/slickSlider.js:1]: all indets must be consistent, use prittier for example. you need to make settings for code formating

File: hw2/assets/sass/_mixins.sass [hunk 4]
```diff
@@ -0,0 +1,30 @@
+@mixin btn
+	font-family: acme, sans-serif
+	display: inline-block
+	text-transform: uppercase
+	height: 40px
```
Comment (Maxim-Bondar) [inline hw2/assets/sass/_mixins.sass:5]: don't hardcode button height

File: hw2/assets/sass/_mixins.sass [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+	border-radius: 20px
+	font-size: 12px
+	letter-spacing: 2px
+	outline: none
+	
+@mixin btnTransparent
+	background-color: transparent
+	color: $accent-green-color
+
+@mixin btnPaddingX($paddingX)
```
Comment (Maxim-Bondar) [inline hw2/assets/sass/_mixins.sass:19]: What is a profit from mexin with one css property?

File: hw2/assets/sass/_mixins.sass [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+	border: 2px solid $accent-green-color
+	background-color: $accent-green-color
+	box-shadow: 0px 6px 12px 0px rgba(127, 127, 127, 0.3)
+	color: #fff
+	border-radius: 20px
+	font-size: 12px
+	letter-spacing: 2px
+	outline: none
+	
+@mixin btnTransparent
```
Comment (Maxim-Bondar) [inline hw2/assets/sass/_mixins.sass:15]: such mixin was created for changing button background , as I understood from mixin name, but you have an extra `color` property inside

File: hw2/assets/sass/_vars.sass [hunk 7]
```diff
@@ -0,0 +1,25 @@
+$gray-text-color: #7F7F7F
```
Comment (Maxim-Bondar) [inline hw2/assets/sass/_vars.sass:1]: color variables shouldn't contain strict dependency with color value

File: hw2/assets/sass/main.sass [hunk 8]
```diff
@@ -0,0 +1,1490 @@
+@import "fonts"
```
Comment (Maxim-Bondar) [inline hw2/assets/sass/main.sass:1]: We don't use SASS , please use SCSS

File: hw2/assets/sass/main.sass [hunk 9]
```diff
@@ -0,0 +1,1490 @@
...
+@import "vars"
+@import "mixins"
+
+*
```
Comment (Maxim-Bondar) [inline hw2/assets/sass/main.sass:5]: Reset and general styles must be in a separete file

File: hw2/assets/sass/main.sass [hunk 10]
```diff
@@ -0,0 +1,1490 @@
...
+					margin: 0 auto
+					margin-bottom: 16px
+					user-select: none
+
+					&:hover
+						cursor: pointer
+						color: $white-color
+						background-color: $accent-green-color
+
+				.get-skout-btn
```
Comment (Maxim-Bondar) [inline hw2/assets/sass/main.sass:181]: don't need create different classes for each buttons, you need to have one common button class and if styles are different create specific class for such button. 
Also you have a strict dependency to button text in class name, what if the text will be changed?

File: hw2/assets/sass/main.sass [hunk 11]
```diff
@@ -0,0 +1,1490 @@
...
+					@include btnPaddingX(54px)
+
+				.log-in-btn
+					@include btnPaddingX(22px)
+
+		#menu__toggle:checked ~ .menu__btn > span
```
Comment (Maxim-Bondar) [inline hw2/assets/sass/main.sass:187]: this implementation must be made by JS

File: hw2/assets/sass/main.sass [hunk 12]
```diff
@@ -0,0 +1,1490 @@
...
+	&__credo
+		display: flex
+		flex-direction: column
+		justify-content: center
+		align-items: center
+
+		.credo__phrase
+			margin-bottom: 20px
+
+			p
```
Comment (Maxim-Bondar) [inline hw2/assets/sass/main.sass:231]: use class selector insted of tag

File: hw2/assets/sass/main.sass [hunk 13]
```diff
@@ -0,0 +1,1490 @@
...
+				&::before
+					position: absolute
+					top: 5px
+					left: 0
+					content: ''
+					width: 2px
+					height: 75%
+					background-color: $white-color
+
+		.why-skout-btn
```
Comment (Maxim-Bondar) [inline hw2/assets/sass/main.sass:250]: read comment for buttons above

File: hw2/assets/sass/main.sass [hunk 14]
```diff
@@ -0,0 +1,1490 @@
...
+
+			&:hover
+				color: $white-color
+				background-color: $accent-green-color
+				border-color: $accent-green-color
+
+	&__slogan
+		text-transform: uppercase
+		font-family: actor, sans-serif
+		font-size: 60px
```
Comment (Maxim-Bondar) [inline hw2/assets/sass/main.sass:265]: font-size also should be taken from variables

File: hw2/assets/sass/main.sass [hunk 15]
```diff
@@ -0,0 +1,1490 @@
...
+		letter-spacing: 0px
+		text-align: left
+		margin-bottom: 20px
+
+	.about
+
+		.container
+			display: grid
+
+		// .fade-in
```
Comment (Maxim-Bondar) [inline hw2/assets/sass/main.sass:291]: remove comments

File: hw2/assets/sass/main.sass [hunk 16]
```diff
@@ -0,0 +1,1490 @@
...
+			font-family: aclonica, sans-serif
+			font-size: 13px
+			line-height: 24px
+			letter-spacing: 0px
+			text-align: left
+
+		&__buttons
+			text-align: center
+
+			.learn-more-btn
```
Comment (Maxim-Bondar) [inline hw2/assets/sass/main.sass:320]: read about button above

File: hw2/assets/sass/main.sass [hunk 17]
```diff
@@ -0,0 +1,1490 @@
...
+					color: $tip-text-color
+					background-color: $white-color
+					border: 1px solid $light-gray-color
+					border-radius: 7px
+					padding: 8px 3px
+
+		.more-tips-btn
+			@include btnPaddingX(65px)
+			display: block
+			margin: 20px auto 80px auto
```
Comment (Maxim-Bondar) [inline hw2/assets/sass/main.sass:401]: magrin shouldn't be in button itself, it must be on a parent element

File: hw2/assets/sass/main.sass [hunk 18]
```diff
@@ -0,0 +1,1490 @@
...
+
+			&:hover
+				color: $accent-green-color
+				border-color: $accent-green-color
+				background-color: $white-color
+
+	.problems
+		background: -webkit-linear-gradient(45deg, rgb(152, 207, 195), rgb(248, 248, 250), rgb(255, 255, 255))
+		background: -moz-linear-gradient(45deg, rgb(152, 207, 195), rgb(248, 248, 250), rgb(255, 255, 255))
+		background: linear-gradient(45deg, rgb(152, 207, 195), rgb(248, 248, 250), rgb(255, 255, 255))
```
Comment (Maxim-Bondar) [inline hw2/assets/sass/main.sass:411]: gradient also can be moved to the variables file

File: hw2/assets/sass/main.sass [hunk 19]
```diff
@@ -0,0 +1,1490 @@
...
+		&__slide
+			padding: 35px 20px
+			border-radius: 5px
+			background-color: $white-color
+			box-shadow: 0px 12px 24px 0px rgba(229, 229, 229, 1)
+			max-width: 260px
+			transition: $transition-time
+
+			&:hover
+				// background-color: transparent
```
Comment (Maxim-Bondar) [inline hw2/assets/sass/main.sass:526]: remove comment

File: hw2/assets/sass/main.sass [hunk 20]
```diff
@@ -0,0 +1,1490 @@
...
+
+			.city-title
+				font-family: aclonica, sans-serif
+				font-size: 16px
+				font-style: normal
+				font-weight: 400
+				line-height: 36px
+				letter-spacing: 0px
+
+			nav
```
Comment (Maxim-Bondar) [inline hw2/assets/sass/main.sass:814]: use classes instead of tag

File: hw2/assets/sass/main.sass [hunk 21]
```diff
@@ -0,0 +1,1490 @@
...
+			img
+				max-width: 105px
+
+
+// ===============================================================
+// ==========================MEDIA================================
+// ===============================================================
+
+
+@media screen and (min-width: 768px)
```
Comment (Maxim-Bondar) [inline hw2/assets/sass/main.sass:862]: please format your code

File: hw2/assets/sass/main.sass [hunk 22]
```diff
@@ -0,0 +1,1490 @@
...
+					top: 5px
+					left: 0
+					content: ''
+					width: 2px
+					height: 75%
+					background-color: $white-color
+
+		&__slogan
+			padding: 200px 30px 100px
+			// padding-top: 200px
```
Comment (Maxim-Bondar) [inline hw2/assets/sass/main.sass:992]: rwmove comments

File: hw2/index.html [hunk 23]
```diff
@@ -0,0 +1,592 @@
...
+	<link rel="stylesheet" href="assets/css/main.css">
+</head>
+<body>
+	
+	<header class="main-header" id="mainHeader">
+		<div class="container main-header__container">
+			<a href="#" class="main-header__logo"></a>
+			<div class="main-header__nav">
+				<div class="hamburger-menu">
+				  <input id="menu__toggle" type="checkbox" />
```
Comment (Maxim-Bondar) [inline hw2/index.html:17]: it must be a button and implemented by JS

File: hw2/index.html [hunk 24]
```diff
@@ -0,0 +1,592 @@
...
+				  <label class="menu__btn" for="menu__toggle">
+				    <span></span>
+				  </label>
+				  <div class="menu__box">
+				  	<nav class="menu__nav" id="mobileMenu">
```
Comment (Maxim-Bondar) [inline hw2/index.html:22]: id also should have consistent style as class names

File: hw2/index.html [hunk 25]
```diff
@@ -0,0 +1,592 @@
...
+				  			<li><a class="menu__item active" href="#">why skout</a></li>
+						    <li><a class="menu__item" href="#">products</a></li>
+						    <li><a class="menu__item" href="#">partners</a></li>
+						    <li><a class="menu__item" href="#">company</a></li>
+						    <li><a class="menu__item" href="#">tips & tools</a></li>
+						    <li><a class="menu__item" href="#">blog</a></li>
+				  		</ul>
+					  </nav>
+					  <div class="menu__buttons">
+					  	<button class="btn btn-transparent get-skout-btn">get skout</button>
```
Comment (Maxim-Bondar) [inline hw2/index.html:33]: regarding to the class naming read comment which I have left in styles file for button classes

File: hw2/index.html [hunk 26]
```diff
@@ -0,0 +1,592 @@
...
+					<nav>
+						<ul>
+							<li><a href="#">Blog</a></li>
+							<li><a href="#">Company</a></li>
+							<li><a href="#">Terms & Conditions</a></li>
+							<li><a href="#">Privacy Poacy</a></li>
+							<li><a href="#">Careers</a></li>
+						</ul>
+					</nav>
+					<p class="main-phone">
```
Comment (Maxim-Bondar) [inline hw2/index.html:509]: it is also a list element

File: hw2/index.html [hunk 27]
```diff
@@ -0,0 +1,592 @@
...
+						Call the SOC <br>
+						<a class="phone-number" href="tel:18558384500">1-855-838-4500</a>
+					</p>
+				</div>
+
+				<div class="footer-contacts">
+					<div class="footer-block block-1">
```
Comment (Maxim-Bondar) [inline hw2/index.html:516]: block-1 is bad class name, what if such blocks will be 10 items? will you create 10 classes for each section?
also such blocks have similar stiles? why do you need such class?

**Review by Maxim-Bondar — CHANGES_REQUESTED** [review]
COMMON
- rename all class where you have strict dependency with content inside
- don't need to create separete classes for each button
- avoid such classes as .block-1 etc.

File: hw2/assets/js/fadeIn.js (hunk 1 above)
Comment (VladyslavPolushyn) [reply hw2/assets/js/fadeIn.js:1]: Done

File: hw2/assets/js/mainMenu.js (hunk 2 above)
Comment (VladyslavPolushyn) [reply hw2/assets/js/mainMenu.js:19]: Done

File: hw2/assets/js/slickSlider.js (hunk 3 above)
Comment (VladyslavPolushyn) [reply hw2/assets/js/slickSlider.js:1]: Don't understand what exactly is wrong with indentation formatting?

File: hw2/assets/sass/_mixins.sass (hunk 4 above)
Comment (VladyslavPolushyn) [reply hw2/assets/sass/_mixins.sass:5]: Done

File: hw2/assets/sass/_mixins.sass (hunk 5 above)
Comment (VladyslavPolushyn) [reply hw2/assets/sass/_mixins.sass:19]: I changed the mixin a bit. The point is not to repeat padding-left and padding-right. And if you use a shorthand record, then you need to specify also padding-top and padding-bottom.

File: hw2/assets/sass/_mixins.sass (hunk 6 above)
Comment (VladyslavPolushyn) [reply hw2/assets/sass/_mixins.sass:15]: It was realy unnecessary. Done.

File: hw2/assets/sass/_vars.sass (hunk 7 above)
Comment (VladyslavPolushyn) [reply hw2/assets/sass/_vars.sass:1]: Done

File: hw2/assets/sass/main.sass (hunk 9 above)
Comment (VladyslavPolushyn) [reply hw2/assets/sass/main.sass:5]: Done

File: hw2/assets/sass/main.sass (hunk 12 above)
Comment (VladyslavPolushyn) [reply hw2/assets/sass/main.sass:231]: Done

File: hw2/assets/sass/main.sass (hunk 15 above)
Comment (VladyslavPolushyn) [reply hw2/assets/sass/main.sass:291]: Done

File: hw2/assets/sass/main.sass (hunk 17 above)
Comment (VladyslavPolushyn) [reply hw2/assets/sass/main.sass:401]: Done

File: hw2/assets/sass/main.sass (hunk 18 above)
Comment (VladyslavPolushyn) [reply hw2/assets/sass/main.sass:411]: Done

File: hw2/assets/sass/main.sass (hunk 19 above)
Comment (VladyslavPolushyn) [reply hw2/assets/sass/main.sass:526]: Done

File: hw2/assets/sass/main.sass (hunk 21 above)
Comment (VladyslavPolushyn) [reply hw2/assets/sass/main.sass:862]: Done

File: hw2/assets/sass/main.sass (hunk 22 above)
Comment (VladyslavPolushyn) [reply hw2/assets/sass/main.sass:992]: Done

File: hw2/index.html (hunk 24 above)
Comment (VladyslavPolushyn) [reply hw2/index.html:22]: Done

File: hw2/index.html (hunk 26 above)
Comment (VladyslavPolushyn) [reply hw2/index.html:509]: Done

File: hw2/index.html (hunk 27 above)
Comment (VladyslavPolushyn) [reply hw2/index.html:516]: I agree.
Moreover these were unnecessary classes

File: hw2/index.html (hunk 25 above)
Comment (VladyslavPolushyn) [reply hw2/index.html:33]: Done

File: hw2/assets/sass/main.sass (hunk 10 above)
Comment (VladyslavPolushyn) [reply hw2/assets/sass/main.sass:181]: Done

File: hw2/assets/sass/main.sass (hunk 16 above)
Comment (VladyslavPolushyn) [reply hw2/assets/sass/main.sass:320]: Done

File: hw2/assets/sass/main.sass (hunk 14 above)
Comment (VladyslavPolushyn) [reply hw2/assets/sass/main.sass:265]: Done

File: hw2/assets/sass/main.sass (hunk 13 above)
Comment (VladyslavPolushyn) [reply hw2/assets/sass/main.sass:250]: Done

File: hw2/assets/sass/main.sass (hunk 8 above)
Comment (VladyslavPolushyn) [reply hw2/assets/sass/main.sass:1]: Got it.
I will take this into account in the following works.

File: hw2/assets/sass/main.sass (hunk 11 above)
Comment (VladyslavPolushyn) [reply hw2/assets/sass/main.sass:187]: I will take this into account in the following works. This block was html / css and I thought it was not necessary to use js.

File: hw2/index.html (hunk 23 above)
Comment (VladyslavPolushyn) [reply hw2/index.html:17]: I will take this into account in the following works. This block was html / css and I thought it was not necessary to use js.

**Review by Maxim-Bondar — APPROVED** [review]

**Pull request merged**